containers-contrib-readonly/src/Map/Contrib/Laws.hs

```diff
@@ -1,8 +1,77 @@
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE ViewPatterns #-}
+
 module Map.Contrib.Laws
        ( singletonFromList
+       , nullEmpty
+       , nullImpliesZeroSize
+       , nonZeroSizeImpliesNotNull
+       , sizeIsNatural
+       , emptyZeroSized
+       , singletonOneSized
+       , memberEmptyFalse
+       , memberSingletonSame
+       , newMemberYieldsValidValue
+       , lookupDefaultEmpty
+       , listToSingleton
+       , keysOfSingleton
+       , elemsOfSingleton
+       , lookupMatchMember
        ) where
 
-import Map (Key, fromList, singleton)
+import Map
+import Prelude hiding (lookup, null)
+import           Test.QuickCheck
+
+nullEmpty :: Bool
+nullEmpty = null empty
+
+nullImpliesZeroSize :: Key k => Gen [(k, v)] -> Gen Property
```

Comment:
Hmm, having generator for `Property` looks a bit counter-intuitive 🤔

Reply:
I agree. It typechecks because of the fancy class `Testable` but it's not intuitive, I'll try to find a better way :)